deps/rabbit/src/rabbit_amqp_session.erl

```diff
@@ -2719,8 +2719,12 @@ ensure_source(#'v1_0.source'{
         _ ->
             exit_not_implemented("Dynamic source not supported: ~tp", [Source0])
     end;
+ensure_source(undefined, _, _, _, _, _, _, _, _) ->
+    {error, source_address_required};
```

Comment:
Be more specific. The error here is not `source_address_required`, it's `source_required` since the entire source is undefined.

I know it's just an error being returned, but if you could add a test case via the Erlang client or .NET or Java client that hits this branch of the code, that would be appreciated too. For example I added various test cases asserting for specific errors, one example can be found in https://github.com/rabbitmq/rabbitmq-server/blob/4ba98d8c6c65b357d0f1721b5aabc80b9560b88f/deps/rabbit/test/amqp_address_SUITE.erl#L216-L230